### PR TITLE
perfdata added to file open list

### PR DESCRIPTION
**Pull request by GitMensch**
along with a group that includes it and moving out to a separate function again

fixing the side issue mentioned in #440

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -279,10 +278,18 @@ void MainWindow::closeEvent(QCloseEvent* event)
     QMainWindow::closeEvent(event);
 }
 
+QString MainWindow::getOpenDataFileName()
```
Comment (milianw) [inline src/mainwindow.cpp:281]: please remove the `get` prefix

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -279,10 +278,18 @@ void MainWindow::closeEvent(QCloseEvent* event)
...
+{
+    const QString filter = tr("Hotspot data Files (perf*.data perf.data.* *.perfparser);;"
```
Comment (milianw) [inline src/mainwindow.cpp:283]: I'd prefer just `auto filter = ...`

File: src/mainwindow.cpp (hunk 1 above)
Comment (milianw) [reply src/mainwindow.cpp:281]: and make this method const

**Review by milianw — CHANGES_REQUESTED** [review]
please reword your commit message, it's pretty confusing. it would be better to clearly state that you are adding the `*.perfdata` pattern to the file open filter list

File: src/mainwindow.cpp [hunk 3]
```diff
@@ -279,10 +278,18 @@ void MainWindow::closeEvent(QCloseEvent* event)
     QMainWindow::closeEvent(event);
 }
 
+const QString MainWindow::openDataFileName()
```
Comment (milianw) [inline src/mainwindow.cpp:281]: const the method, not the return type

and I think it would be better to rename this function to `queryOpenDataFile`

**Review by milianw — CHANGES_REQUESTED** [review]

File: src/mainwindow.cpp (hunk 3 above)
Comment (GitMensch) [reply src/mainwindow.cpp:281]: renaming sounds good - how do you want the signature to be - and how is this function different to other utility functions in this source file?

Comment (milianw) [reply src/mainwindow.cpp:281]:
```
QString MainWindow::queryOpenDataFile() const
```

I don't quite get what you mean by that question? are you saying there are utility functions that can be const which are not marked as const? fix that then please

**Review by milianw — APPROVED** [review]
code lgtm, let's wait for the CI and then this can be merged

**Comment by milianw** [comment]
Argh sorry, it cannot be const b/c we need to pass `this` as a non-const `QWidget *` parent... sorry please remove the `const` again!

File: .gitpod.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+image: workspace-c
```
Comment (milianw) [inline .gitpod.yml:1]: unrelated file?

**Review by milianw — CHANGES_REQUESTED** [review]

File: .gitpod.yml (hunk 4 above)
Comment (GitMensch) [reply .gitpod.yml:1]: totally

**Review by milianw — APPROVED** [review]
